### PR TITLE
[CORE-129] updates to get idle coeffs surfaced, also surface provider ID

### PR DESCRIPTION
**Pull request by ameijer**
## What does this PR change?
* Surfaces idle coefficients and provider IDs

## Does this PR relate to any other PRs?
* yes, updates https://github.com/opencost/opencost/pull/1796 so that can be closed

## How will this PR impact users?
* introduces a new includeProportionalAssetResourceCosts query param that can be used to return proportional costs for instances or clusters. other API behavior is unaffected 

## Does this PR address any GitHub or Zendesk issues?
* Closes https://github.com/opencost/opencost/pull/1796

## How was this PR tested?
* confirmed no regressions using local build. confirmed numbers returned looked appropriate based on manual test

## Does this PR require changes to documentation?
* there is an additional query param that can be documented 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* is a specific customer ask

### CODE REVIEW

File: pkg/costmodel/assets.go
Comment (michaelmdresser) [inline pkg/costmodel/assets.go:11]: @nikovacevic is this going to become the canonical `ComputeAssets`, also used by KCM?

File: pkg/costmodel/costmodel.go
Comment (michaelmdresser) [inline pkg/costmodel/costmodel.go]: I'm not a fan of silently changing parameters that are incorrectly combined. I'd rather a helpful error be thrown. I can be convinced otherwise, though!

File: pkg/costmodel/costmodel.go
Comment (michaelmdresser) [inline pkg/costmodel/costmodel.go]: These cases look identical! Am I missing something?

Comment (ameijer) [reply pkg/costmodel/costmodel.go]: ah good catch - we had been intending to fix that - changed those to be cluster properties

File: pkg/costmodel/costmodel.go
Comment (ameijer) [reply pkg/costmodel/costmodel.go]: I like the idea. A helpful error message to serve as a guardrail. @nikovacevic thoughts?

**Review by michaelmdresser — APPROVED** [review]
Approving to unblock. Looks like you're missing a DCO signoff on the latest commit.

I'd really like to see an improvement on the defaulting thread, but I certainly won't block the PR on it :slightly_smiling_face:

**Review by michaelmdresser — APPROVED** [review]

**Pull request merged**
